### DISCUSSION TITLE
Consume sceptre as a library

### DISCUSSION THREAD

**Discussion by alex-harvey-z3q** [cloudreach/sceptre#1046] [Q&A]
I wish to consume Sceptre as a library in a Python project. I note however that comments in the code recommend not importing the `sceptre.cli` module. Is there documentation on how to import Sceptre as a library, is it even really possible given the existing code organisation, and are there any code examples?

**Comment by ykhalyavin** [comment]
We render all / some templates for later validation with cfn-lint / cfn_nag / etc with a simple custom script like:

```
from sceptre.context import SceptreContext
from sceptre.plan.plan import SceptrePlan
from sceptre.helpers import normalise_path
...
context = SceptreContext(
    project_path=project_path.as_posix(),
    command_path=sceptre_group,
    user_variables=user_variables,
    ignore_dependencies=True
)
plan = SceptrePlan(context)
...
for stack, template_body in plan.generate().items():
    ...
```

One reason for a custom script -- **_sceptre generate asd/xyz.yaml_** CLI will print out multiple yaml templates alltogether if there're dependencies (not sure if it's fixed).
Anoher reason -- we want to generate all stack configs * templates, or a specific group and save them to separate files.

@alexharv074  hope it helps

**Comment by zaro0508** [comment]
thanks for the info @ykhalyavin.  Wondering if you could create a PR to add that info to the Sceptre docs?

**Reply by ngfgrant** [reply]
Yeh this is an oversight on my part - one of the big aims of v2 was to make sceptre easier to use as a library.

I did do something for the migration guide: https://github.com/sceptre/project/wiki/Migration-Guide:-V1-to-V2#concepts-and-terminology

but it doesn't look like that made it into the official docs.
